pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -764,29 +800,14 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 
 	// Step 6: start containers in podContainerChanges.ContainersToStart.
 	for _, idx := range podContainerChanges.ContainersToStart {
-		container := &pod.Spec.Containers[idx]
-		startContainerResult := kubecontainer.NewSyncResult(kubecontainer.StartContainer, container.Name)
-		result.AddSyncResult(startContainerResult)
-
-		isInBackOff, msg, err := m.doBackOff(pod, container, podStatus, backOff)
-		if isInBackOff {
-			startContainerResult.Fail(err, msg)
-			klog.V(4).Infof("Backing Off restarting container %+v in pod %v", container, format.Pod(pod))
-			continue
-		}
+		start("container", &pod.Spec.Containers[idx])
```

Comment:
Why discarding the returned error?

Reply:
That's the current behavior: start the container, log any errors, move on to the next container. The logging is now done in `start()`

I should mention that the only behavior here that I (intentionally) changed was that the log spam filtering for `ErrImagePullBackOff` was previously only applied to regular containers and not init containers.